packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -0,0 +1,3369 @@
+import { ApprovalController } from '@metamask/approval-controller';
+import { ControllerMessenger } from '@metamask/base-controller';
+import {
+  ApprovalType,
+  BUILT_IN_NETWORKS,
+  NetworkType,
+} from '@metamask/controller-utils';
+import {
+  NetworkController,
+  NetworkClientType,
+} from '@metamask/network-controller';
+import nock from 'nock';
+import type { SinonFakeTimers } from 'sinon';
+import { useFakeTimers } from 'sinon';
+
+import { advanceTime } from '../../../tests/helpers';
+import { mockNetwork } from '../../../tests/mock-network';
+import {
+  ETHERSCAN_TRANSACTION_BASE_MOCK,
+  ETHERSCAN_TRANSACTION_RESPONSE_MOCK,
+  ETHERSCAN_TOKEN_TRANSACTION_MOCK,
+  ETHERSCAN_TRANSACTION_SUCCESS_MOCK,
+} from './helpers/EtherscanMocks';
+import { TransactionController } from './TransactionController';
+import type { TransactionMeta } from './types';
+import { TransactionStatus, TransactionType } from './types';
+import { getEtherscanApiHost } from './utils/etherscan';
+import * as etherscanUtils from './utils/etherscan';
+
+const ACCOUNT_MOCK = '0x6bf137f335ea1b8f193b8f6ea92561a60d23a207';
+const ACCOUNT_2_MOCK = '0x08f137f335ea1b8f193b8f6ea92561a60d23a211';
+const ACCOUNT_3_MOCK = '0xe688b84b23f322a994a53dbf8e15fa82cdb71127';
+const infuraProjectId = '341eacb578dd44a1a049cbc5f6fd4035';
+
+const networkClientConfiguration = {
+  type: NetworkClientType.Infura,
+  network: NetworkType.goerli,
+  chainId: BUILT_IN_NETWORKS[NetworkType.goerli].chainId,
+  infuraProjectId,
+  ticker: BUILT_IN_NETWORKS[NetworkType.goerli].ticker,
+} as const;
+
+const sepoliaNetworkClientConfiguration = {
+  type: NetworkClientType.Infura,
+  network: NetworkType.sepolia,
+  chainId: BUILT_IN_NETWORKS[NetworkType.sepolia].chainId,
+  infuraProjectId,
+  ticker: BUILT_IN_NETWORKS[NetworkType.sepolia].ticker,
+} as const;
+
+const mainnetNetworkClientConfiguration = {
+  type: NetworkClientType.Infura,
+  network: NetworkType.mainnet,
+  chainId: BUILT_IN_NETWORKS[NetworkType.mainnet].chainId,
+  infuraProjectId,
+  ticker: BUILT_IN_NETWORKS[NetworkType.mainnet].ticker,
+} as const;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const newController = async (options: any = {}) => {
```

Comment:
can we make this better?

Reply:
ughh.. there's a lot of other type errors hiding behind this. I'm not sure this is worth fixing

Reply:
abandon ship